packages/cli-e2e/__tests__/react.specs.ts

```diff
@@ -7,80 +7,116 @@ import {captureScreenshots, getNewBrowser, openNewPage} from '../utils/browser';
-              )
-            ) {
-              resolve();
-            }
-          });
-        }),
-      ]);
-
-      const startServerProcess = processManager.spawn('npm', ['run', 'start'], {
-        cwd: getProjectPath(projectName),
-      });
-      await new Promise<void>((resolve) => {
-        startServerProcess.stdout.on('data', async (data) => {
+import {join} from 'path';
+import {renameSync} from 'fs-extra';
+
+describe('ui:create:react', () => {
+  let browser: Browser;
+  let buildProcessManager: ProcessManager;
+  let page: Page;
+  const oldEnv = process.env;
+  const projectName = `${process.env.GITHUB_ACTION}-react-project`;
+  // TODO: CDX-90: Assign a dynamic port for the search token server on all ui projects
+  const clientPort = '3000';
+  const searchPageEndpoint = `http://localhost:${clientPort}`;
+  const tokenProxyEndpoint = `http://localhost:${clientPort}/token`;
+
+  const buildApplication = async (processManager: ProcessManager) => {
+    const proc = setupUIProject(processManager, 'ui:create:react', projectName);
+
+    return Promise.race([
+      new Promise<void>((resolve) => {
+        proc.on('exit', async () => {
+          resolve();
+        });
+      }),
+      new Promise<void>((resolve) => {
+        proc.stdout.on('data', (data) => {
           if (
-            /You can now view .*-react-project in the browser/.test(
+            /Happy hacking !/.test(
               stripAnsi(data.toString()).replace(/\n/g, '')
             )
           ) {
             resolve();
           }
         });
+      }),
+    ]);
+  };
+
+  const startApplication = async (processManager: ProcessManager) => {
+    const proc = processManager.spawn('npm', ['run', 'start'], {
+      cwd: getProjectPath(projectName),
+    });
+
+    return new Promise<void>((resolve) => {
+      proc.stdout.on('data', async (data) => {
+        if (
+          /You can now view .*-react-project in the browser/.test(
+            stripAnsi(data.toString()).replace(/\n/g, '')
+          )
+        ) {
+          resolve();
+        }
       });
-    }, 15 * 60e3);
+    });
+  };
+
+  const deactivateEnvironmentFile = () => {
+    const pathToEnv = getProjectPath(projectName);
+    renameSync(join(pathToEnv, '.env'), join(pathToEnv, '.env.disabled'));
+  };
+
+  const restoreEnvironmentFile = () => {
+    const pathToEnv = getProjectPath(projectName);
+    renameSync(join(pathToEnv, '.env.disabled'), join(pathToEnv, '.env'));
+  };
+
+  beforeAll(async () => {
+    buildProcessManager = new ProcessManager();
+    browser = await getNewBrowser();
+    await buildApplication(buildProcessManager);
+  }, 10 * 60e3);
+
+  beforeEach(async () => {
+    jest.resetModules();
+    process.env = {...oldEnv};
+    page = await openNewPage(browser, page);
+  });
+
+  afterEach(async () => {
+    await captureScreenshots(browser);
+  });
+
+  afterAll(async () => {
+    process.env = oldEnv;
+    await buildProcessManager.killAllProcesses();
```

Comment:
I would:
 - Refactor `buildProcessManager` and `serverProcessManager` in a processManagers array.
 - Ensure that all processManagers are closed in the global afterAll
 - Kill all processes of the buildProcess at the end of the beforeAll. He doesn't make sense for it to be still running after the beforeAll.
 - Kill all processes of the server process in each afterAll sections (which I think you already do)
 
 The top-level afterAll is a catch 'em all clause in case something goes haywire, but at least we're sure of ourselves.
 The refactor into an array is to simplify adding other processes in the future.

Reply:
I agree with the 2 last points. But not too much with the 2 firsts.
Killing multiple time the processes of the same ProcessManager will likely lead to `Error: kill ESRCH`.

Reply:
> Killing multiple times the processes of the same ProcessManager will likely lead to `Error: kill ESRCH`.

Hm, no actually.
The processManager unregister its registered processes as they exit, and the 'killing function' is awaitable (and should be awaited).
Therefore, if this happens, it means there's an issue either in the exiting of a processor in the processManager behaviour.

In either case, it would fail legitimately and should be 'fixed' and not ignored.

The most important thing in a test is to have a trustable setup phase (i.e. we know that the test environment is as we designed/thought it).
When you're not able to trust your setup phase, that means tests result could be due to unforeseen conditions that are not applicable for our end-user, leading to a false positive or false negative.

If that'd be the case, the tests would be worthless.

Reply:
Fixed `killAllProcesses`.
It was not removing a process from the processes list once it was killed